tools/integration_tests/implicit_dir/delete_test.go

```diff
@@ -0,0 +1,140 @@
+	// Clean the mountedDirectory before running test.
+	setup.CleanMntDir()
+
+	implicit_and_explicit_dir_setup.CreateImplicitDirectory()
+	explicitDirPath := path.Join(setup.MntDir(), implicit_and_explicit_dir_setup.ImplicitDirectory, implicit_and_explicit_dir_setup.ImplicitSubDirectory, ExplicitDirInImplicitSubDir)
+
+	operations.CreateDirectoryWithNFiles(NumberOfFilesInExplicitDirInImplicitSubDir, explicitDirPath, PrefixFileInExplicitDirInImplicitSubDir, t)
+
+	dirPath := path.Join(setup.MntDir(), implicit_and_explicit_dir_setup.ImplicitDirectory)
+
+	implicit_and_explicit_dir_setup.RemoveAndCheckIfDirIsDeleted(dirPath, implicit_and_explicit_dir_setup.ImplicitDirectory, t)
+}
+
+// Directory Structure
+// testBucket/explicitDirectory                                                                   -- Dir
+// testBucket/explictFile                                                                         -- File
+// testBucket/explicitDirectory/fileInExplicitDir1                                                -- File
+// testBucket/explicitDirectory/fileInExplicitDir2                                                -- File
+// testBucket/explicitDirectory/implicitDirectory                                                 -- Dir
+// testBucket/explicitDirectory/implicitDirectory/fileInImplicitDir1                              -- File
+// testBucket/explicitDirectory/implicitDirectory/implicitSubDirectory                            -- Dir
+// testBucket/explicitDirectory/implicitDirectory/implicitSubDirectory/fileInImplicitDir2         -- File
+func TestDeleteImplicitDirInExplicitDir(t *testing.T) {
+	// Clean the mountedDirectory before running test.
+	setup.CleanMntDir()
+
+	implicit_and_explicit_dir_setup.CreateImplicitDirectoryInExplicitDirectory(t)
+
+	dirPath := path.Join(setup.MntDir(), implicit_and_explicit_dir_setup.ExplicitDirectory, implicit_and_explicit_dir_setup.ImplicitDirectory)
+
+	implicit_and_explicit_dir_setup.RemoveAndCheckIfDirIsDeleted(dirPath, implicit_and_explicit_dir_setup.ImplicitDirectory, t)
+}
+
+// Directory Structure
+// testBucket/explicitDirectory                                                                   -- Dir
+// testBucket/explictFile                                                                         -- File
+// testBucket/explicitDirectory/fileInExplicitDir1                                                -- File
+// testBucket/explicitDirectory/fileInExplicitDir2                                                -- File
+// testBucket/explicitDirectory/implicitDirectory                                                 -- Dir
+// testBucket/explicitDirectory/implicitDirectory/fileInImplicitDir1                              -- File
+// testBucket/explicitDirectory/implicitDirectory/implicitSubDirectory                            -- Dir
+// testBucket/explicitDirectory/implicitDirectory/implicitSubDirectory/fileInImplicitDir2         -- File
+func TestDeleteExplicitDirWithImplicitSubDir(t *testing.T) {
```

Comment:
What's the difference b/w this test and the above test (TestDeleteImplicitDirInExplicitDir)?

Reply:
Above test is removing implicit directory from explicit directory
e.g
imp1
imp1/expsub
imp1/expsub/impsub2

remove(impsub2)

This test is removing explicit directory

e.g
imp1
imp1/expsub
imp1/expsub/impsub2

remove(expsub)

Reply:
Okay. Directory structure is same, and we are deleting different object.

We should rename it to -> TestDeleteExplicitDirContainingImplicitSubDir